### ISSUE TITLE
Lates 

### ISSUE THREAD

**Issue by vitalisb** [bigtreetech/BIGTREETECH-SKR-mini-E3#247]
According to the latest marlin code compiled 512k file firmware.bin (bltouch and safe home) does not flash to SKR e3 mini v1.0

**Comment by txt4nk** [comment]
Reply with your printer srtup and ill compile one and see if i get the same results. If i get something that will work for you ill post it here.

**Comment by krakow2000** [comment]
Got the same problem, even if i wanna compile marlin for v1.0 i got error.

**Comment by txt4nk** [comment]
Are you using SKR's firmware version or vanilla marlin?

**Comment by txt4nk** [comment]
also, what software are you using to compile?

**Comment by krakow2000** [comment]
Atom but in visual got the same problem. Yes, SKR firmware version for 1.0

**Comment by txt4nk** [comment]
What printer and mods do you have? I'll try making one and see if I get the same results. If it works, I'll post the firmware bin file for you

**Comment by krakow2000** [comment]
SKR e3 mini v1.0 tmc2209, BLTouch (-45, -15, 0)

**Comment by txt4nk** [comment]
My E3 mini worked amazingly when I used vanilla marlin. I'm going to try that for you.

Can you look at the board and verify the processor model? I've seen RCT6 and RET6's floating around

**Comment by txt4nk** [comment]
Also, is it an ender 3? This board is cross compatible with the CR10 and Ender 5, so I need to know the model for bed sizes and such. If you have a 5, do you have the standard 5 or 5 pro? (z screw changed on the pro model, required 800 steps instead of 400)

**Comment by krakow2000** [comment]
Ender 3, RET6

**Comment by txt4nk** [comment]
alright, standby.

**Comment by txt4nk** [comment]
Below is the vanilla marlin 2.0.x bugfix full folder (for your safe keeping and tweaking if need-be) and also the firmware.bin (in a zip file so I could upload it) compiled from the linked project.

[Marlin-bugfix-2.0.x_krakow2000.zip](https://github.com/bigtreetech/BIGTREETECH-SKR-mini-E3/files/4638710/Marlin-bugfix-2.0.x_krakow2000.zip)
[firmware.zip](https://github.com/bigtreetech/BIGTREETECH-SKR-mini-E3/files/4638711/firmware.zip)

**Comment by krakow2000** [comment]
ty, will inform you after test :)

**Comment by txt4nk** [comment]
Looking forward to it!

**Comment by krakow2000** [comment]
soooo slow auto home :) rest work correct for now

**Comment by kirelam** [comment]
Hi,

I have a few settings to speed up the homing and probing. Will mail them.

Regards,
Kirelam

Op za 16 mei 2020 18:54 schreef krakow2000 <notifications@github.com>:

**Comment by txt4nk** [comment]
haha yeah I didn't fine tune any of that, was more focused on making sure the firmware would flash before spending more time on it. I can speed things up.

**Comment by krakow2000** [comment]
:)

**Comment by txt4nk** [comment]
Awesome. Does that project file compile for you? I'd like to know what caused your firmware to not work but mine did.. could you zip the folder you were working in and upload it? I'd like to inspect for differences.

**Comment by krakow2000** [comment]
yes

**Comment by txt4nk** [comment]
I have a suspicion that it is the processor setting in the platformio.ini or the board name in the configuration.h

**Comment by krakow2000** [comment]
nah... board name correct, change in planformio STM32F103RE_btt_USB but still nothing :)

**Comment by txt4nk** [comment]
I didn't use the USB processor, I used STM32F103RE_btt. I wonder if that difference really causes the problem

**Comment by krakow2000** [comment]
i can show you old btt build

**Comment by txt4nk** [comment]
yes please

**Comment by krakow2000** [comment]
https://drive.google.com/open?id=1Tk-Qumot-EEqdfgCvtyqdHdpbZWcdtDa

**Comment by txt4nk** [comment]
The processor was wrong, the serial ports were wrong and the board is correct.

I used STM32F103RE_btt, whereas they used STM32F103RC_bigtree_NOUSB
![image](https://user-images.githubusercontent.com/65292849/82127190-50292e80-9777-11ea-91ca-23ed4dbed61a.png)

Primary serial port is 2, secondary serial port is -1. They got them backwards.
![image](https://user-images.githubusercontent.com/65292849/82127318-36d4b200-9778-11ea-852f-13ae5126c28b.png)
